File: schemaorggo/menusection.go
package schemaorggo

import "encoding/json"

// MenuSection see : https://schema.org/MenuSection
type MenuSection struct {
	CreativeWork

	typeContext

	// HasMenuItem see : https://schema.org/hasMenuItem
	// A food or drink item contained in a menu or menu section.
	// types : MenuItem
	HasMenuItem *MenuItem `json:"hasMenuItem,omitempty"`

	// HasMenuSection see : https://schema.org/hasMenuSection
	// A subgrouping of the menu (by dishes, course, serving time period, etc.).
	// types : MenuSection
	HasMenuSection *MenuSection `json:"hasMenuSection,omitempty"`
}

func (v MenuSection) MarshalJSONWithTypeContext() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "MenuSection"

	return json.Marshal(v)
}

func (v *MenuSection) MarshalJSON() ([]byte, error) {
	if v == nil {
		return []byte("null"), nil
	}

	v.C = "http://schema.org"
	v.T = "MenuSection"

	return json.Marshal(*v)
}

File: schemaorgphp/classes/menu_section.php
<?php

class MenuSection extends CreativeWork implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'MenuSection';
	
	/**
	 * A food or drink item contained in a menu or menu section.
	 * see : https://schema.org/hasMenuItem
	 * @var \MenuItem|\MenuItem[]
	 */
	public var $has_menu_item;
	
	/**
	 * A subgrouping of the menu (by dishes, course, serving time period, etc.).
	 * see : https://schema.org/hasMenuSection
	 * @var \MenuSection|\MenuSection[]
	 */
	public var $has_menu_section;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'MenuSection'
		);
		
		$serialized = so_json_serialize( $this->has_menu_item );
		if ( ! empty( $serialized ) ) {
			$out['hasMenuItem'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->has_menu_section );
		if ( ! empty( $serialized ) ) {
			$out['hasMenuSection'] = $serialized;
		}
		
		return $out;
	}
}
